common/src/main/java/com/microsoft/identity/common/internal/controllers/NativeAuthMsalController.kt

```diff
@@ -39,17 +86,735 @@ class NativeAuthMsalController : BaseNativeAuthController() {
     companion object {
         private val TAG = NativeAuthMsalController::class.java.simpleName
     }
-    
+
+    /**
+     * Acts as the starting point for the Sign In flow.
+     * Starts off with the sign in initiate call, and returns any error results if produced.
+     * If successful, continues ot the sign in challenge call, again returning any error results if produced.
+     * If the challenge endpoint returns a password required result,
+     * a call to the token endpoint will be made with the password provided in the SignInStartCommandParameters,
+     * again returning any error results, or returning the token if the call succeeds.
+     * If a token is acquired, it is also cached locally as part of this flow.
+     */
+    fun signInStart(parameters: SignInStartCommandParameters): SignInStartCommandResult {
+        LogSession.logMethodCall(TAG, "${TAG}.signInStart")
+
+        try {
+            val oAuth2Strategy = createOAuth2Strategy(parameters)
+
+            val usePassword = parameters is SignInStartUsingPasswordCommandParameters
+
+            var parametersWithScopes: SignInStartUsingPasswordCommandParameters? = null
+            if (usePassword) {
+                val mergedScopes = addDefaultScopes(parameters.scopes)
+                parametersWithScopes = CommandUtil.createSignInStartCommandParametersWithScopes(
+                    parameters as SignInStartUsingPasswordCommandParameters,
+                    mergedScopes
+                )
+            }
+
+            val initiateApiResult = performSignInInitiateCall(
+                oAuth2Strategy = oAuth2Strategy,
+                parameters = parameters // Downcast, because we don't need the password in this API call
```

Comment:
How do you know that? This doesn't seem to be in an `else` block that corresponds with the previous if

Reply:
Know what? I don't understand the question.

Reply:
I have refactored this method slightly.